### ISSUE TITLE
Support the HTML autocomplete attribute for accessibility

### ISSUE THREAD

**Issue by avernet** [orbeon/orbeon-forms#5083]
Documentation:

- [WCAG 2.1](https://www.w3.org/TR/WCAG21/#input-purposes)
- [MDN](https://developer.mozilla.org/en-US/docs/Web/HTML/Attributes/autocomplete)

[+1 from customer](https://3.basecamp.com/3600924/buckets/2192969/messages/4417906841)

**Comment by avernet** [comment]
[+1 from customer](https://3.basecamp.com/3600924/buckets/23878674/messages/5289266047)

**Comment by VanessaTsang888** [comment]
@avernet I just clicked on the link from your comment on May 4 but it is broken. The error msg is "Sorry, we don't have anything to show you on this page". Please can you give us an update?

![image](https://github.com/orbeon/orbeon-forms/assets/48724277/e3f11242-c93f-4728-92ea-05d1ae94e5f7)

**Comment by avernet** [comment]
@VanessaTsang888 Those +1's were from other customers (hence the message you got from Basecamp), and I see that we were missing a +1 from you, so I added it below.

[+1 from customer](https://3.basecamp.com/3600924/buckets/3893142/messages/6097519081)

**Comment by avernet** [comment]
Resolutions:

- For XBL components: we will not provide support for this in the XForms engine; instead, each XBL component will copy the `autocomplete' attribute as needed (or more, as in the case of the Dropdown Date and the Field Date for the `bday` case).
- TBD: Make a list of XBL controls that need to support this, determine how the Control Settings knows whether to show the autocomplete dropdown (information provided by the XBL control or can reuse something we already have), and which values to show in the dropdown (for example, for dates, only `bday` makes sense).

**Comment by ebruchez** [comment]
- [ ] Form Builder: UI in Control Settings to set the `autocomplete` attribute
    - [ ] basic implementation
    - [ ] considering filtering allowed values by control type (for example  a date shouldn't be an email)
        - show value if set on control, but don't allow selecting
    - [ ] Email: should have automatically `email`, etc.
- [ ] XForms controls: forward the `autocomplete` to the correct native form element
- [ ] XBL Components: case by case forward attribute

**Comment by obruchez** [comment]
- [ ] do not expose/enable autocomplete select for incompatible controls
- [ ] make the list of autocomplete values more readable (indent them?)
- [ ] add default autocomplete values to US address template
- [ ] investigate whether shipping/billing is supported by browsers
- [ ] filter autocomplete values based on control type (?) (+ show value if set on control, but don't allow selecting)
- [ ] add support for multi-input controls (date/time, etc.)

**Comment by obruchez** [comment]
- [ ] remove "off", move last 3 values to "General" group
